File: Competitions/Codeforces Round 856 (Div. 2)/C.cpp
#include <iostream>

using namespace std;


int main() {
    int amount;
    int lenth;

    cin >> amount;

    for (int step = 0; step < amount; ++step) {
        cin >> lenth;

        int array[lenth];

        for (int place = 0; place < lenth; ++place) {
            cin >> array[place];
        }

        for (int cut = 0; cut < lenth; ++cut) {
            int cut_last = cut;
            int res = array[cut_last];
            int d = 1;
            int m = 1;

            while (cut_last) {
                if (array[cut_last - 1] >= d + 1) {
                    res *= array[cut_last - 1];
                    ++d;
                    --cut_last;
                    ++m;
                }
            }

            cout << m << " ";
        }
    }

    return 0;
}

File: Competitions/Codeforces Round 856 (Div. 2)/C.py
def maximum_cost(array, current_start_place):
    cost = 1
    multiplication = array[-1]
    d = 1

    while current_start_place:
        if multiplication * array[current_start_place - 1] / (d+1) >= multiplication / d:
            cost += 1
            multiplication *= array[current_start_place]
            d += 1
            current_start_place -= 1
        else:
            break

    return cost


def main():
    amount = int(input())

    for step in range(amount):
        lenth = int(input())
        array = list(map(int, input().split()))

        for cut in range(1, lenth + 1):
            cutted = array[:cut]

            print(maximum_cost(cutted, cut - 1), end=" ")

        print()


main()
